### PR TITLE
Upgrade to 0.60.4

### PR DESCRIPTION
**Pull request by wswebcreation**
Sorry for the big ass PR, but this one was needed for the following reasons:
- we needed to upgrade to the latest version of React Native to support the latest version of Android
- React Native made some really big changes to the source code (that's why we have that amount of files touched, **please focus only on the `tests` and `src` folders )**
- We needed to update a lot of modules due to changes in them
- a lot of customers are asking on how to automate deeplinking and biometrical login and I can only integrate that when I updated the project

So, basically, everything changed 😆 

I'm on holiday as of now, so take your time. When this is merged I will make new releases for the Sauce Labs users and customers and will upload it here https://github.com/saucelabs/sample-app-mobile/releases

That was it, happy checking!

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,11827 +0,0 @@
-{
```
Comment (rslobojan-saucelabs) [inline package-lock.json]: Shouldn't package-lock.json remain? Otherwise our builds won't be repeatable since the precise versions of the packages used won't be the same.

File: README.md [hunk 2]
```diff
@@ -5,6 +5,9 @@
 In this repository you will find our Sauce Labs sample-app. You can use it as a sample app for testautomation on your local machine or in our Real Device Cloud.
 The latest version of the iOS and Android app can be found [here](https://github.com/saucelabs/sample-app-mobile/releases)
 
+> **NOTE<br>**
+> Make sure to use yarn instead of NPM. NPM gives a lot of problems currently with React Native. How to setup yarn can be found [here](https://yarnpkg.com/lang/en/docs/install/#mac-stable)
```
Comment (rslobojan-saucelabs) [inline README.md]: What kinds of problems? And where should we look to know when they are resolved and we can start using NPM again?

File: LICENSE [hunk 3]
```diff
@@ -1,21 +0,0 @@
-MIT License
```
[the remark this answers is not in the document]
Comment (wswebcreation) [reply LICENSE]: Oops, sorry, will put it back

File: README.md [hunk 4]
```diff
@@ -22,24 +25,25 @@ This is the Sauce Labs Sample Application which is designed to be used from iOS
...
 
 - install homebrew: https://brew.sh/
-- brew install node
-- brew install watchman
-- npm install -g react-native-cli
+- `brew install node`
+- `brew install yarn`
+- `brew install watchman`
+- `yarn global add react-native-cli`
 - install JDK8: https://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html
```
[the remark this answers is not in the document]
Comment (wswebcreation) [reply README.md]: 👍 , will change it

File: README.md [hunk 5]
```diff
@@ -5,6 +5,9 @@
 In this repository you will find our Sauce Labs sample-app. You can use it as a sample app for testautomation on your local machine or in our Real Device Cloud.
 The latest version of the iOS and Android app can be found [here](https://github.com/saucelabs/sample-app-mobile/releases)
 
+> **NOTE<br>**
```
[the remark this answers is not in the document]
Comment (wswebcreation) [reply README.md]: It is put there intentionally, otherwise line 9 will not start on a new line

File: package-lock.json (hunk 1 above)
Comment (wswebcreation) [reply package-lock.json]: Because we need to use yarn and not npm I removed the npm `package-lock.json` file, it can also not be uploaded because it's in the `.gitignore` file

File: README.md (hunk 2 above)
Comment (wswebcreation) [reply README.md]: Good remarks and questions. I don't have exact answers, only from what I've seen with the migration.

- React Natve 0.60.x now uses Pods for linking dependencies, from what I've seen with NPM is that is messes up this linking 9 out of 10 times
- Installing other dependencies are mostly recommended to install with yarn (not only for React Native but also with ReactJS)
- It costed me around a day to get everything working with NPM, but with yarn I didn't had any problems migrating to RN 0.60
- yarn is faster

and I can go on

**Comment by wswebcreation** [comment]
@rslobojan-saucelabs 

I fixed the changes and updated the comment, can you please take a look and if you agree resolve the last remark?
I'll then merge it this week and create a new release

File: README.md (hunk 2 above)
Comment (rslobojan-saucelabs) [reply README.md]: Interesting - good to know! I think it would make sense to expand upon the notes above in the README since npm is often the default choice for building Node projects - I think you can basically copy/paste what you said above into the doc and it'll make things a lot clearer for the reader (and also the note about .gitinore'ing package-lock.json so it's clear there's a particular reason for that choice)

Comment (wswebcreation) [reply README.md]: Will do, again thanks for your time, appreciate it a lot!
